modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2491,6 +2491,44 @@ def _callable_func(self, func, axis, *args, **kwargs):
     groupby_prod = GroupByReduce.register("prod")
     groupby_sum = GroupByReduce.register("sum")
 
+    def _mean_agg_map(dfgb, **kwargs):
+        kwargs["min_count"] = 1
+        result = dfgb.sum(**kwargs)
+        divisor = dfgb.count()
+        divisor.set_axis(
+            ["__mean_agg_size_column__" + x for x in divisor.columns],
+            axis=1,
+            inplace=True,
+        )
+        result = pandas.concat([result, divisor], axis=1, copy=False)
+        return result
+
+    def _mean_agg_reduce(dfgb, **kwargs):
+        kwargs["min_count"] = 1
+        result = dfgb.sum(**kwargs)
+        divirgent = result[
+            [x for x in result.columns if not x.startswith("__mean_agg_size_column__")]
```

Comment:
Thinking again about this, probably a best approach would be to add a new multiindex level which would distinguish between original columns and result of `.size()` operation. What do you think? I am not sure whether to do it this way in all cases or only in case of MultIndex columns.

Reply:
`mean` is `df.sum() / df.count()` (or `df.groupby.sum()`/`df.groupby.count()`), and `size` is just `len` so I don't think it would be correct. Each column has an independent `count`

Reply:
Yes, it should be `count()` instead of `size()` in my comment above (I edited it too for clarity). The comment itself was not about using `size()`, but was about distinguishing new columns from original without adding a string prefix to their names in a multiindex case.

Reply:
I see, I have a prototype of this here: https://github.com/modin-project/modin/pull/1902/files